app/controllers/chat_channels_controller.rb

```diff
@@ -76,4 +76,34 @@ def notification_settings
       render_json_error(membership)
     end
   end
+
+  def for_tag
+    params.require(:tag_name)
+
+    tag = Tag.find_by(name: params[:tag_name])
+    raise Discourse::NotFound unless tag
+
+    render_channel_for_chatable(
```

Comment:
I think perhaps you need to add a check around tag permissions for your `ensure_can_see_chat_channel` guardian extension, like we do in core for the `Tag#show` endpoint?

https://github.com/discourse/discourse/blob/7ce927eaa84bfaa9e72bd2f371625f2f95f6c42e/app/controllers/tags_controller.rb#L344-L346

Reply:
Good call, moved some logic to the guardian that was only in the `ChatChannelFetcher`, to secure this 👍

Reply:
Sweet!